src/modules/reports/reports.controller.ts

```diff
@@ -0,0 +1,26 @@
+import {
+  BadRequestException,
+  Body,
+  Controller,
+  Get,
+  Post,
+  Query,
+} from '@nestjs/common';
+import { ReportsService } from './reports.service';
+import { ApiBody } from '@nestjs/swagger';
+import { DatesDTO } from './models/reports.dto';
+
+@Controller('reports')
+export class ReportsController {
+  constructor(private readonly reportsService: ReportsService) {}
+
+  @Post('patientsDates')
+  @ApiBody({ type: DatesDTO })
+  async findPatientsBetweenDates(@Body() dates: DatesDTO): Promise<Report[]> {
+    console.log('Received dates:', dates);
+    return this.reportsService.findBetweenDates(
```

Comment:
its a better practice to pass all the object to the service and in the service just get the values i mean
` return this.reportsService.findBetweenDates(body)`